File: springboot/src/main/java/com/edufelip/meer/dto/AuthDtos.java
package com.edufelip.meer.dto;

public class AuthDtos {
    public record LoginRequest(String email, String password) {}
    public record SignupRequest(String name, String email, String password) {}
    public record GoogleLoginRequest(String client, String idToken) {}
    public record AppleLoginRequest(String provider, String idToken, String authorizationCode, String client) {}
    public record RefreshRequest(String refreshToken) {}
    public record ForgotPasswordRequest(String email) {}

    public record UserDto(String id, String name, String email) {}

    public record AuthResponse(String token, String refreshToken, UserDto user) {}
}

File: springboot/src/main/kotlin/com/edufelip/meer/dto/AuthDtos.kt
package com.edufelip.meer.dto

data class LoginRequest(
    val email: String,
    val password: String
)

data class SignupRequest(
    val name: String,
    val email: String,
    val password: String
)

data class GoogleLoginRequest(
    val client: String,
    val idToken: String? = null
)

data class RefreshRequest(
    val refreshToken: String
)

data class AppleLoginRequest(
    val provider: String,
    val idToken: String,
    val authorizationCode: String? = null,
    val client: String
)

data class ForgotPasswordRequest(
    val email: String
)

data class UserDto(
    val id: String,
    val name: String?,
    val email: String?
)

data class AuthResponse(
    val token: String,
    val refreshToken: String?,
    val user: UserDto
)
